server/mqtt.go

```diff
@@ -598,11 +597,6 @@ func validateMQTTOptions(o *Options) error {
 	if mo.Port == 0 {
 		return nil
 	}
-	// We have to force the server name to be explicitly set. There are conditions
-	// where we need a unique, repeatable name.
-	if o.ServerName == _EMPTY_ {
```

Comment:
This might be ok for single server mode, but not clustered mode. In clustred mode this is required.

Reply:
Peer names, which need to be consistent, are a derived hash from server names. So in clustered mode this condition has to hold. So we need to check if we are in single server mode, or not using JetStream, although I thought it was required for MQTT.

Reply:
JS is required for MQTT; I am 0/5 if it makes any sense to remove the name requirement for single server only. @bruth?

Reply:
There is a js helper function to test if in single server mode.

Reply:
@levb The observation was that enabling jetstream (without MQTT) in single-server mode does not require an explicit server name, so it felt unnecessary only because MQTT is enabled.

Reply:
K, pushed a fix, please review. @derekcollison this is unfortunately in the context of options, so no convenience methods. Can you please see if I need to add `(... || o.Gateway.Port != 0 || o.LeafNode.Port != 0)`?

Reply:
Look at what the js helper uses to determine..

Reply:
Let's do same thing.

Reply:
I did, unfortunately it just checks that the run-time object is not nil, that it has initialized. This one, https://github.com/nats-io/nats-server/blob/main/server/jetstream_cluster.go#L724?

Reply:
```
// Determines if this server is in standalone mode, meaning no routes or gateways.
func (s *Server) standAloneMode() bool {
	opts := s.getOpts()
	return opts.Cluster.Port == 0 && opts.Gateway.Port == 0
}
```

Reply:
ah! thx!

Reply:
@derekcollison Can you please confirm